### PR TITLE
Cleanup xblock

### PR DESCRIPTION
**Pull request by stvstnfrd**
# Overview
Here's some extra cleanup that came out while porting the changes over the `qualtricssurvey` and `grademe-button`.

# Test Instructions
- Checkout the branch
- `make test && make runserver`

# TODO
- [x] Compile static assets
- [x] Lint all files
- [x] Pass all tests
- [ ] Code Reviewer 1: @caesar2164

### CODE REVIEW

File: .tx/config [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [main]
 host = https://www.transifex.com
 
-[xblocks.image-modal]
+[stanford-openedx-xblocks.image-modal]
```
Comment (stvstnfrd) [inline .tx/config]: This was outdated from what was on Transifex.

File: Makefile [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/make -f
 module_root := ./imagemodal
-css_files := $(patsubst %.less, %.less.css, $(wildcard ./$(module_root)/public/*.less))
+css_files := $(patsubst %.less, %.css, $(wildcard ./$(module_root)/public/*.less))
```
Comment (stvstnfrd) [inline Makefile]: CSS files used to be named like `*.less.css`;
this removes the `.less` part.

File: README.rst [hunk 3]
```diff
@@ -15,8 +16,12 @@ the full-size version of the image.
...
 
 
-Installation, system administrator
-----------------------------------
+Installation
+------------
+
+
+System Administrator
+~~~~~~~~~~~~~~~~~~~~
```
Comment (stvstnfrd) [inline README.rst]: Make these actually be a further subheading.

File: imagemodal/mixins/fragment.py [hunk 4]
```diff
@@ -2,14 +2,53 @@
...
+        Build a context dictionary to render the student view
+
+        This should generally be overriden by child classes.
+        """
+        context = context or {}
+        context = dict(context)
+        return context
+
+    @XBlock.supports('multi_device')
+    def student_view(self, context=None):
```
Comment (stvstnfrd) [inline imagemodal/mixins/fragment.py]: Provide a default student view that adds normal css/js files into the default template file.
You can override/extend paths.
This moves more logic out of each XBlock view and into the mixin.
Eventually, the mixin will be split out into a shared repo.

File: imagemodal/mixins/fragment.py [hunk 5]
```diff
@@ -22,12 +61,12 @@ def build_fragment(
         """
         Creates a fragment for display.
         """
-        template = 'templates/' + template
         context = context or {}
         css = css or []
         js = js or []
         rendered_template = ''
         if template:
+            template = 'templates/' + template
```
Comment (stvstnfrd) [inline imagemodal/mixins/fragment.py]: This was a bug.

File: imagemodal/views.py [hunk 6]
```diff
@@ -22,13 +21,22 @@ class ImageModalViewMixin(
...
     loader = ResourceLoader(__name__)
+    static_css_extra = [
+        URL_FONT_AWESOME_CSS,
+        'view.css',
+    ]
+    static_js = [
+        'draggabilly.pkgd.min.js',
+        'view.js',
+    ]
+    static_js_init = 'ImageModalView'
```
Comment (stvstnfrd) [inline imagemodal/views.py]: Configure our custom static paths.

File: imagemodal/views.py [hunk 7]
```diff
@@ -22,13 +21,22 @@ class ImageModalViewMixin(
...
 
-    @XBlock.supports('multi_device')
-    def student_view(self, context=None):
+    def provide_context(self, context=None):
```
Comment (stvstnfrd) [inline imagemodal/views.py]: All this view really needs to do is build up a context dictionary.

File: imagemodal/xblocks.py [hunk 8]
```diff
@@ -1,5 +1,5 @@
 """
-This is the core logic for the Image Modal XBlock
+This is the core logic for the XBlock
```
Comment (stvstnfrd) [inline imagemodal/xblocks.py]: Generalizing to make it easier/more consistent across xblocks.

File: setup.py [hunk 9]
```diff
@@ -1,7 +1,6 @@
 """
 A full-screen image modal XBlock
 """
-import json
```
Comment (stvstnfrd) [inline setup.py]: This wasn't being used any longer.

File: imagemodal/views.py [hunk 10]
```diff
@@ -22,13 +21,22 @@ class ImageModalViewMixin(
     """
 
     loader = ResourceLoader(__name__)
+    static_css_extra = [
+        URL_FONT_AWESOME_CSS,
+        'view.css',
+    ]
+    static_js = [
+        'draggabilly.pkgd.min.js',
```
Comment (caesar2164) [inline imagemodal/views.py]: Does this work with the imported version from google? (Or wherever it's coming from)

Comment (stvstnfrd) [reply imagemodal/views.py]: I don't follow... Google?

Comment (caesar2164) [reply imagemodal/views.py]: Didn't we stop loading that file locally and use an online version?

Comment (stvstnfrd) [reply imagemodal/views.py]: > Didn't we stop loading that file locally and use an online version?

No.

I'm concerned that you approved this PR that refactors how we build fragments and pull in static assets, but don't follow how we build fragments and pull in static assets.
It's alright if you don't understand, but please don't approve code you don't follow.

I'd like for you to spend a little while going over this code and tracing the execution.
After you've spent some time trying to grok it, we can sit down and you can walk me through how you think it works and I can fill in any blanks.

Comment (caesar2164) [reply imagemodal/views.py]: Ah nvmd, we're importing this from a node_module. I got confused for a second.

**Comment by caesar2164** [comment]
Other funky error:
```
(devstack_stanford) [giuliog@mbp ~/devstack_vagrant/stanford_ginkgo/xblocks/xblock-image-modal (cleanup *)]$ make clean
coverage erase
make: coverage: Command not found
make: *** [Makefile:27: clean] Error 127
```

**Comment by stvstnfrd** [comment]
@caesar2164 Ready for review.

**Pull request merged**